tests/arch/arm/arm_irq_advanced_features/src/arm_shared_irq.c

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2022 The Chromium OS Authors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/ztest.h>
+#include <zephyr/arch/cpu.h>
+#include <zephyr/arch/arm/aarch32/cortex_m/cmsis.h>
```

Comment:
This all test should be moved to `tests/kernel/interrupt` and properly `ifdef`-fed to work only for AArch32 for now.

Reply:
I moved my test to tests/kernel/interrupt and the AArch32 tests can be moved in a separate PR.